common/djangoapps/student/views.py

```diff
@@ -735,6 +736,8 @@ def dashboard(request):
         'course_programs': course_programs,
         'disable_courseware_js': True,
         'xseries_credentials': xseries_credentials,
+        'use_ecommerce_payment_flow': EcommerceService().is_enabled(),
```

Comment:
The problem is that the ``course_modes`` view is for one course and here it is for all the courses in the dashboard, therefor I’ve placed a condition in the template (which is the only one that iterates through the courses) to check if the mode has a SKU as well.